features/algebraic-types-coding.feature

```diff
@@ -83,17 +84,18 @@ Feature: Outputting Algebraic Types
       - (instancetype)initWithCoder:(NSCoder *)aDecoder
       {
         if ((self = [super init])) {
-          _subtype = [aDecoder decodeIntegerForKey:kSubtypeKey];
-          switch (_subtype) {
-            case _SimpleADTSubtypesFirstSubtype: {
-              _firstSubtype_firstValue = [aDecoder decodeObjectForKey:kFirstSubtypeFirstValueKey];
-              _firstSubtype_secondValue = [aDecoder decodeIntegerForKey:kFirstSubtypeSecondValueKey];
-              break;
-            }
-            case _SimpleADTSubtypesSecondSubtype: {
-              _secondSubtype_something = [aDecoder decodeBoolForKey:kSecondSubtypeSomethingKey];
-              break;
-            }
+          _codedSubtype = [aDecoder decodeObjectForKey:kCodedSubtypeKey];
```

Comment:
It can. I tried to reuse as much of the existing code as possible, but will change.